apps/frontend/src/pages/CreateAccount.tsx

```diff
@@ -1,5 +1,206 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  TextField,
+  Typography,
+  FormHelperText,
+  Link,
+} from "@mui/material";
+import martletPlaceLogo from "../images/martletplace-logo.png";
+import { colors } from "../styles/colors";
+import { useStyles } from "../styles/pageStyles";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+
 const CreateAccount = () => {
-  return <h1>CreateAccount</h1>;
+  const classes = useStyles();
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [emailError, setEmailError] = useState("");
+  const [usernameError, setUsernameError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  const isFormEmpty = !email || !name || !username || !password;
+
+  const navigate = useNavigate();
+
+  const handleCreateAccount = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // Regex for UVic email validation
+    const emailFormat = /^[^@]+@uvic\.ca$/;
+
+    // Regex for username validation
+    const usernameFormat = /^[a-zA-Z]{1,20}$/;
+
+    // Regex for password validation
+    const passwordFormat =
+      /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[\W_])[A-Za-z\d\W_]{8,}$/;
+
+    if (!emailFormat.test(email)) {
```

Comment:
[Regex for emails is hard](https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression). Is there any reason to use regex if the only thing we need to check is that it ends in `@uvic.ca` (and if it's an invalid format they won't get a confirmation email, which is confirmation enough). This would be my recommendation considering that the original regex also permits invalid emails (`h i@uvic.ca` is accepted when it clearly isn't a valid email)


```suggestion
    if (!emailFormat.endsWith("@uvic.ca")) {
```

Reply:
Yeah that's fair. The email submission box already says that it must be a valid uvic email 🤷‍♀️ 

I've defined the input box for email as of type="email" so the MUI component will handle the errors.

<img width="541" alt="Screenshot 2024-06-07 at 2 00 02 PM" src="https://github.com/UVicMartletplace/martletplace/assets/72957712/ead19ff8-7afb-47a5-bec1-a928e534c519">